var/spack/repos/builtin/packages/fortrilinos/package.py

```diff
@@ -72,3 +74,28 @@ def cmake_args(self):
             self.define('ForTrilinos_EXAMPLES', self.run_tests),
             self.define('ForTrilinos_TESTING', self.run_tests),
         ]
+
+    examples_src_dir = 'example/test-installation'
+
+    @run_after('install')
+    def setup_smoke_tests(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources([self.examples_src_dir])
+
+    def test(self):
+        test_build_dir = join_path(self.install_test_root,
+                                   self.examples_src_dir, 'build')
+        mkdirp(test_build_dir)
+        with working_dir(test_build_dir):
+            cmake(
+                self.define('CMAKE_PREFIX_PATH', self.prefix),
+                self.define('CMAKE_CXX_COMPILER', self.compiler.cxx),
+                self.define('CMAKE_Fortran_COMPILER', self.compiler.fc),
+                '..'
+            )
+            make()
+
+        with working_dir(test_build_dir):
+            self.run_test('ctest', ['-V'], [], installed=False,
+                          purpose='test: installation')
```

Comment:
I recommend cleaning up after the test (e.g., make('clean')) so the test is ready (or clean) to be re-built days or weeks later.

Reply:
I think it could be useful to have the build files around in case of a failing test that needs to be diagnosed. More motivation to run in a temporary directory that can be preserved. I'll add a `shutil.rmtree` for the temporary build dir.

Reply:
Understandable.

Spack already creates a test "stage" directory, `self.test.stage`, which is created under (see https://spack.readthedocs.io/en/latest/packaging_guide.html#configuring-the-test-stage-directory).

So an alternative is to copy the files from `self.install_test_root` before you build them in the test stage directory.  Or you could perform a `make clean` (or equivalent) before running the tests where they were copied at build time.

Reply:
Thanks @tldahlgren  for pointing me to `self.test_suite.stage`. I've updated the test to build inside a subdirectory there and point to the read-only source files in the install.